fabric-events-interaction-v0/src/testmod/java/net/fabricmc/fabric/test/event/interaction/InteractionEventsTest.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.test.event.interaction;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import net.minecraft.block.Blocks;
+import net.minecraft.util.ActionResult;
+
+import net.fabricmc.api.ModInitializer;
+import net.fabricmc.fabric.api.event.player.AfterBreakBlockCallback;
+import net.fabricmc.fabric.api.event.player.BeforeBreakBlockCallback;
+
+public class InteractionEventsTest implements ModInitializer {
+	public static final Logger LOGGER = LogManager.getLogger("InteractionEventsTest");
+
+	@Override
+	public void onInitialize() {
+		BeforeBreakBlockCallback.EVENT.register(((pos, state, entity, block) -> {
+			return block == Blocks.BEDROCK ? ActionResult.FAIL : ActionResult.PASS;
+		}));
+
+		AfterBreakBlockCallback.EVENT.register(((pos, state, entity, block) -> {
+			LOGGER.info("Block broken at " + pos.toShortString());
```

Comment:
`toShortString()` is client-only.

Reply:
also remove ur debug statement :tiny_potato: